internal/bgp/frr/config.go

```diff
@@ -35,7 +41,15 @@ router bgp {{.MyASN}}
   neighbor {{.Addr}} port {{.Port}}
 {{- end }}
 {{range $n := .Neighbors -}}
-{{range .Advertisements }}
+{{/* no bgp default ipv4-unicast prevents peering if no address families are defined. We declare an ipv4 one for the peer to make the pairing happen */}}
+{{- if eq (len .Advertisements) 0}}
+  address-family ipv4 unicast
+    neighbor {{$n.Addr}} activate
+  address-family ipv6 unicast
+    neighbor {{$n.Addr}} activate
+  exit-address-family
```

Comment:
is this for testing ? we can't have bgp peer with having addresspool config if that happened it should have caught eariler or should have been caught if that isn't the case ?

Reply:
That's to mimic the same behaviour we have now.
Took me a while to find it. Without a service, frr won't establish a connection with the peer because of the `no bgp default ipv4-unicast`.

"By default, only the IPv4 unicast address family is announced to all neighbors. Using the ‘no bgp default ipv4-unicast’ configuration overrides this default so that all address families need to be enabled explicitly."
So without this, FRR will complain of no address families configured.

Reply:
that was my question I did noticed too if no service no bgp neighbors I couldn't answer if this is an ok thing or not why do we care of having bgp peering if we don't have any network to advertise , I don't know hence asked that queston

Reply:
I am not sure it's right or not, but the current implementation does that and it did not seem bad to replicate that.
Also, that would be the default frr behaviour if the `no bgp default ipv4-unicast` parameter (which is needed to allow both ipv4 and ipv6 to co-exist) wasn't added.